wsdl/ver20/media/wsdl/media.wsdl

```diff
@@ -1300,6 +1327,184 @@ IN NO EVENT WILL THE CORPORATION OR ITS MEMBERS OR THEIR AFFILIATES BE LIABLE FO
+					</xs:annotation>
+					</xs:element>
+					<xs:element name="Gain" type="xs:int" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation> Optional Gain to control volume of the audio clip when it is played.</xs:documentation>
+					</xs:annotation>
+					</xs:element>
+					<xs:element name="ScheduleToken" type="tt:ReferenceToken" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation> Optional Schedule Token associated with the audio clip configuration. Audio clip can be played when configured schedule is triggered.</xs:documentation>
+					</xs:annotation>
+					</xs:element>
+					<xs:element name="PlayStatus" type="xs:boolean" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation> Optional PlayStatus to describe the audio clip is currently playing or not. This parameter is read-only.</xs:documentation>
+					</xs:annotation>
+					</xs:element>
+					<xs:any namespace="##any" processContents="lax" minOccurs="0" maxOccurs="unbounded"/>   <!-- first ONVIF then Vendor -->
+				</xs:sequence>
+				<xs:anyAttribute processContents="lax"/>
+			</xs:complexType>
+			
+			<xs:element name="GetAudioClips">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="Token" type="tt:ReferenceToken" minOccurs="0">
+							<xs:annotation>
+								<xs:documentation>Audio Clip Token.</xs:documentation>
+							</xs:annotation>
+						</xs:element>
+					</xs:sequence>
+				</xs:complexType>
+			</xs:element>			
+			<xs:element name="GetAudioClipsResponse">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="AudioClipsConfiguration" type="tr2:AudioClipConfiguration" minOccurs="0" maxOccurs="unbounded">
+							<xs:annotation>
+								<xs:documentation>Audio clip configurations exists in the device.</xs:documentation>
+							</xs:annotation>
+						</xs:element>
+					</xs:sequence>
+				</xs:complexType>
+			</xs:element>
+			
+			<xs:element name="AddAudioClip">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="AudioClipConfiguration" type="tr2:AudioClipConfiguration">
+							<xs:annotation>
+								<xs:documentation>Audio Clip Configuration to add.</xs:documentation>
+							</xs:annotation>
+						</xs:element>
+					</xs:sequence>
+				</xs:complexType>
+			</xs:element>			
+			<xs:element name="AddAudioClipResponse">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="Token" type="tt:ReferenceToken">						
+							<xs:annotation>
+								<xs:documentation>Unique Token associated with the audio clip.</xs:documentation>
+							</xs:annotation>
+						</xs:element>
+						<xs:element name="UploadUri" type="xs:anyURI">						
+							<xs:annotation>
+								<xs:documentation>An URL to which the audio clip may be uploaded.</xs:documentation>
```

Comment:
How is supposed to work with seperate URI? Format, error codes, etc?

Reply:
Media2 spec document is updated with specific http error codes when audio clip upload fails.